.github/workflows/gradle.yml

```diff
@@ -9,25 +9,30 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions:
+      checks: write
+      contents: read
     strategy:
       matrix:
         java: ['11']
     env:
       WORKSPACE: ${{ github.workspace }}
       GRADLE_OPTS: -Xmx1500m -Dfile.encoding=UTF-8
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v2
```

Comment:
The `gradle/actions/setup-gradle` action performs efficient caching of Gradle User Home so `actions/cache` is not needed:
https://github.com/gradle/actions/blob/af6e5767241073b8ab20cc8cf7e95ec63fadc160/docs/setup-gradle.md#why-use-the-setup-gradle-action